nncf/quantization/algorithms/weight_compression/mixed_precision.py

```diff
@@ -85,20 +99,44 @@ def apply(
         """
         self._set_backend_entity(model)
 
-        scores = self._calc_sensitivity(model, graph, weight_params, statistic_points)
-        num_all_weights = sum(wp.num_weights for wp in weight_params)
-
-        indexes_of_layers_in_ascending_order_of_scores = [
-            i[0] for i in sorted(enumerate(scores), reverse=False, key=lambda x: x[1])
-        ]
-        num_weights_in_4bit = 0
-        for index in indexes_of_layers_in_ascending_order_of_scores:
-            weight_param = weight_params[index]
-            current_ratio = (num_weights_in_4bit + weight_param.num_weights) / num_all_weights
-            if current_ratio >= self._ratio:
-                break
-            weight_param.compression_config = self._primary_config
-            num_weights_in_4bit += weight_param.num_weights
+        flexible_group_size_values = {}
+        valid_weight_params = weight_params
+        if self._enable_flexible_group_size and self._primary_config.group_size != -1:
```

Comment:
I see that `flexible group size` is closely connected with mixed precision, but I don't feel it's responsibility of this class.
What about doing it outside and give a modified `ratio_defining_params`?

Reply:
potentially, can be implemented in `group_quantization.py`: 
https://github.com/openvinotoolkit/nncf/pull/3496#discussion_r2169749898

Reply:
the revised version with lightweight mixed precision part looks good to me